### DISCUSSION TITLE
O.S. Standard Authentication & LDAP

### DISCUSSION THREAD

**Discussion by WillianBR** [filebrowser/filebrowser#2519] [Ideas]
That's for sure a good tool.  I'm testing it here and it's promessing!  The only thing I miss right on the spot, it's a flexible authentication mechanism.  
I think it should came right into the box the support for standard mechanisms:

- Internal database (it has);
- Default OS user authentication ([Linux](https://forum.golangbridge.org/t/check-linux-user-and-password-authenticate/6220/3)/[PAM](https://www.socketloop.com/tutorials/golang-verify-linux-user-password-again-before-executing-a-program-example) and Windows/[Azure](https://learn.microsoft.com/en-us/azure/developer/go/azure-sdk-authentication?tabs=bash) are not that hard);
- LDAP user authentication (it's very common on the companies);

The last one (LDAP), it's not that difficult to find a company using Microsoft AD (not a good LDAP, we can handle).  The standard LDAP still play a good game on this days.    I'm just finishing a project and sooner I'll look at LDAP support.  [Even GO has a library to handle such scenarios](https://github.com/go-ldap/ldap/blob/master/examples_test.go) on project [Basic LDAP v3 functionality for the GO programming language](https://github.com/go-ldap/ldap.git).

I hope sooner, I'll have something working to LDAP!

**Comment by alladin0201** [comment]
wow, that's very interesting, thanks!

**Comment by mastergbc** [comment]
I hope that too!
